### PR TITLE
MULE-13007: Every exported Mule package must contain api (part3)

### PR DESCRIPTION
**Pull request by pablokraan**

### CODE REVIEW

File: core-tests/src/test/java/org/mule/runtime/core/routing/requestreply/AsyncRequestReplyRequesterTestCase.java [hunk 1]
```diff
@@ -221,6 +225,27 @@ public void testMultiple() throws Exception {
...
+          eventBuilder().message(of("message1")).groupCorrelation(new GroupCorrelation(1, null)).build();
+
+      SensingNullMessageProcessor listener = getSensingNullMessageProcessor();
+      mp.setListener(listener);
+      mp.setReplySource(listener.getMessageSource());
+
+      mp.process(event);
+
+      Map<String, Event> responseEvents = mp.getResponseEvents();
+      assertTrue("Response events should be cleaned up.", responseEvents.isEmpty());
```
Comment (elrodro83) [inline core-tests/src/test/java/org/mule/runtime/core/routing/requestreply/AsyncRequestReplyRequesterTestCase.java]: use the empty collection matcher

File: core/src/main/java/org/mule/runtime/core/privileged/processor/AbstractFilteringMessageProcessor.java [hunk 2]
```diff
@@ -4,7 +4,7 @@
  * license, a copy of which has been included with this distribution in the
  * LICENSE.txt file.
  */
-package org.mule.runtime.core.processor;
+package org.mule.runtime.core.privileged.processor;
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/privileged/processor/AbstractFilteringMessageProcessor.java]: by moving ProcessIfStartedMessageProcessor, you can now move this to compatibility too

File: core/src/main/java/org/mule/runtime/core/internal/construct/DefaultFlowBuilder.java [hunk 3]
```diff
@@ -270,7 +270,7 @@ protected void configurePreProcessors(MessageProcessorChainBuilder builder) thro
 
     @Override
     protected void configurePostProcessors(MessageProcessorChainBuilder builder) throws MuleException {
-      builder.chain(new AsyncReplyToPropertyRequestReplyReplier());
+      builder.chain(new SimpleAsyncRequestReplyRequester.AsyncReplyToPropertyRequestReplyReplier());
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/construct/DefaultFlowBuilder.java]: use just the inner class name

File: core/src/main/java/org/mule/runtime/core/internal/processor/strategy/AbstractStreamProcessingStrategyFactory.java [hunk 4]
```diff
@@ -129,7 +129,7 @@ protected String getWaitStrategy() {
     protected AbstractStreamProcessingStrategy(Supplier<Scheduler> ringBufferSchedulerSupplier, int bufferSize, int subscribers,
                                                String waitStrategy, int maxConcurrency) {
       this.subscribers = requireNonNull(subscribers);
-      this.waitStrategy = valueOf(waitStrategy);
+      this.waitStrategy = WaitStrategy.valueOf(waitStrategy);
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/processor/strategy/AbstractStreamProcessingStrategyFactory.java]: static import

File: core/src/main/java/org/mule/runtime/core/internal/processor/strategy/DirectSink.java [hunk 5]
```diff
@@ -39,9 +38,10 @@
   public DirectSink(Function<Publisher<Event>, Publisher<Event>> function, Consumer<Event> eventConsumer) {
     DirectProcessor<Event> directProcessor = create();
     reactorSink =
-        new ReactorSink(directProcessor.serialize().connectSink(), directProcessor.transform(function).doOnError(throwable -> {
-        }).subscribe(),
-                        eventConsumer);
+        new AbstractProcessingStrategy.ReactorSink(directProcessor.serialize().connectSink(),
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/processor/strategy/DirectSink.java]: use just the inner class name

File: core/src/main/java/org/mule/runtime/core/internal/processor/strategy/TransactionAwareProactorStreamProcessingStrategyFactory.java [hunk 6]
```diff
@@ -93,7 +92,7 @@ public ProcessingStrategy create(MuleContext muleContext, String schedulersNameP
     @Override
     public Sink createSink(FlowConstruct flowConstruct, ReactiveProcessor pipeline) {
       Sink proactorSink = super.createSink(flowConstruct, pipeline);
-      Sink syncSink = BLOCKING_PROCESSING_STRATEGY_INSTANCE.createSink(flowConstruct, pipeline);
+      Sink syncSink = BlockingProcessingStrategyFactory.BLOCKING_PROCESSING_STRATEGY_INSTANCE.createSink(flowConstruct, pipeline);
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/processor/strategy/TransactionAwareProactorStreamProcessingStrategyFactory.java]: static import

File: core/src/main/java/org/mule/runtime/core/internal/processor/strategy/TransactionAwareWorkQueueProcessingStrategyFactory.java [hunk 7]
```diff
@@ -61,7 +59,7 @@ protected TransactionAwareWorkQueueProcessingStrategy(Supplier<Scheduler> ioSche
     @Override
     public Sink createSink(FlowConstruct flowConstruct, ReactiveProcessor pipeline) {
       Sink workQueueSink = super.createSink(flowConstruct, pipeline);
-      Sink syncSink = BLOCKING_PROCESSING_STRATEGY_INSTANCE.createSink(flowConstruct, pipeline);
+      Sink syncSink = BlockingProcessingStrategyFactory.BLOCKING_PROCESSING_STRATEGY_INSTANCE.createSink(flowConstruct, pipeline);
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/processor/strategy/TransactionAwareWorkQueueProcessingStrategyFactory.java]: static import

File: core/src/main/java/org/mule/runtime/core/internal/routing/FirstSuccessfulRoutingStrategy.java [hunk 8]
```diff
@@ -4,40 +4,116 @@
...
+        failExceptionCause = ex;
+      }
+      if (!failed) {
+        break;
+      }
+    }
+
+    if (failed) {
+      if (failExceptionCause != null) {
+        throw new RoutingFailedException(createStaticMessage("all processors failed during first-successful routing strategy"),
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/routing/FirstSuccessfulRoutingStrategy.java]: quote (`''`) first-successful

File: core/src/main/java/org/mule/runtime/core/internal/routing/FirstSuccessfulRoutingStrategy.java [hunk 9]
```diff
@@ -4,40 +4,116 @@
...
+                                         failExceptionCause);
+      } else {
+        throw new RoutingFailedException(createStaticMessage("all processors failed during first-successful routing strategy"));
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/routing/FirstSuccessfulRoutingStrategy.java]: Quote first-successfull

File: core/src/main/java/org/mule/runtime/core/internal/routing/FirstSuccessfulRoutingStrategy.java (hunk 8 above)
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/routing/FirstSuccessfulRoutingStrategy.java]: all -> All

File: core/src/main/java/org/mule/runtime/core/internal/routing/FirstSuccessfulRoutingStrategy.java (hunk 9 above)
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/routing/FirstSuccessfulRoutingStrategy.java]: all -> All

File: modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json [hunk 10]
```diff
@@ -1678,7 +1678,7 @@
                 "format": "java",
                 "type": "Object",
                 "annotations": {
-                  "typeId": "org.mule.runtime.core.routing.AggregationStrategy",
+                  "typeId": "org.mule.runtime.core.internal.routing.AggregationStrategy",
```
Comment (elrodro83) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: can internal classes be referenced from the extension model? check with @gsfernandes

Comment (gsfernandes) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: @alegmarra, does this kind of annotation mean that tooling should be able to access this interface to allow users to contribute with a custom strategy?

Comment (pablokraan) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: Moved to api

Comment (alegmarra) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: @gsfernandes in this case, not, is not necessary for tooling to have access to the class, only rt should require it. That said, all the parameters should be of types exposed as part of the api, none internal.

sidenote: with that definition of "custom-aggregation-strategy" tooling is not going to be able to construct the element correctly

**Pull request merged**